src/components/form/field_number/field_number.tsx

```diff
@@ -13,7 +13,7 @@ import { IconType } from '../../icon';
 
 export type EuiFieldNumberProps = InputHTMLAttributes<HTMLInputElement> &
   CommonProps & {
-    value?: number | '';
+    value?: string | number;
```

Comment:
perhaps, if touching this line anyway, it should be updated to match the underlying type?
```suggestion
    value?: InputHTMLAttributes<HTMLInputElement>['value'];
```

by the way, that type is:
```ts
        value?: string | string[] | number;
```

I must admit, I wasn't aware you can pass a string array here.  I'll have to look into that.  [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefvalue) didn't really clear it up for me, which is unusual.

Reply:
![Screenshot_20200309_213554](https://user-images.githubusercontent.com/15232461/76271199-18b97580-624e-11ea-8e8e-423e3da73e7e.png)

![Screenshot_20200309_213338](https://user-images.githubusercontent.com/15232461/76271229-2d960900-624e-11ea-8ae2-fcf5b6d0d23f.png)

I got curious.  It appears that if you pass an array it will just call `toString` which has the effect of deep flattening it.  Interesting..